### ISSUE TITLE
Update check providing old version

### ISSUE THREAD

**Issue by clarkie82** [mRemoteNG/mRemoteNG#1853]
When clicking "Check for Updates", the installed version is higher than the latest version. 
The version I have installed is the latest stable version as per https://mremoteng.org/download
As such, the "latest" version in the check appears to be incorrect, and retrieving that data from https://mremoteng.org/update.txt

Is https://mremoteng.org/update.txt not being updated?

## Possible Solution
Use https://github.com/mRemoteNG/mRemoteNG.github.io/blob/master/_data/releases.json rather than https://mremoteng.org/update.txt for the check

**Comment by clarkie82** [comment]
I changed my mRemoteNG.exe.config to use the below address - this address is as per https://github.com/mRemoteNG/mRemoteNG/wiki/Release-Checklist
```
<applicationSettings>
    <mRemoteNG.Settings>
      <setting name="UpdateAddress" serializeAs="String">
        <value>https://raw.githubusercontent.com/mRemoteNG/mRemoteNG/gh-pages/</value>
      </setting>

```

**Comment by v-karbovnichy** [comment]
I think the `gh-pages` commits does not trigger main website rebuild 
and probably this is related: https://github.com/mRemoteNG/mRemoteNG/commit/acd2460c74c3a0c38d3f2a6d3de4ccde43a047d6

**Comment by heinrichmartin** [comment]
Given that the latest tag is 1.77.1 from 2019-09 (and the latest commit to develop was in 2020-06), is this project maintained? Do you have plans to release the unreleased changelog?

> Use https://github.com/mRemoteNG/mRemoteNG.github.io/blob/master/_data/releases.json

That does not have a 1.77.1 and calls 1.77.0 a pre-release.

**Comment by farosch** [comment]
Closing duplicates #2052 and #1909

**Comment by farosch** [comment]
This will be fixed with #1210

**Comment by simonai1254** [comment]
Just checked if that issue is present with 1.77.2-nb on Windows 10 21H2. 

I don't know how it's handled in the nightly build, though I get 'no updates available', and "Latest Version: 1.77.2\nLatest Version: 1.76.11.40527"

I guess there is an issue with displaying the current version with the tag "latest version", unless it's referring to nighly and stable release channels. In any case it's unclear what is what with the current information.

Further I would suggest to display the information on what update channel is configured (Stable, Prerelease, Nightly)

**Comment by Kvarkas** [comment]
its have some mess there what will be sort tonight - actually meaning is that latest is mean to be stable, but yes perhaps good idea to add info for all 3 channels

**Comment by simonai1254** [comment]
> its have some mess there what will be sort tonight - actually meaning is that latest is mean to be stable, but yes perhaps good idea to add info for all 3 channels

Actually if I would be running Stable I would not care to get the information what version Nightly is at, though I would like to know that I am using the Stable release channel and thereof am using the latest version.
My desired output would look something like this:

// For Nightly Install with nightly release channel set in settings
_Up to date_
_Current Version: 1.77.2 (Nightly Build)_
_Latest Version: 1.77.2_
_Release Channel: Nightly_
_Last Check: 2022-01-12 15:30_

// For Preview Install with Stable release channel set in settings
_Up to date_
_Current Version: 1.77.1 (Preview Build)_
_Latest Version: 1.76.20_
_Release Channel: Stable_
_Last Check: 2022-01-12 15:31_

// For Stable Install with Stable release channel set in settings
_Update Available_
_Current Version: 1.76.11 (Stable)_
_Latest Version: 1.76.20 (Released on 2019-04-12)_
_Release Channel: Stable_
_Change Log: [Link to Release Notes]_
_Last Check: 2022-01-09 10:10_

As far as I can see at the moment there seems to be no tag if the software was installed as "preview" or "nightly", therefore I guess this information can not be displayed.
What is strange though is that the "latest version" is 1.76.11, even though the latest official release is 1.76.20 (at the moment).

I hope you can get this sorted out, thanks for your work :)

**Comment by Kvarkas** [comment]
yes, for nightly it same as preview in means of terms, just not the latest version and acting like a prob so users who like to help with testing can start mikling they fingers on it. but its for experienced users so technically it maybe named as Development channel or Alfa version, Pre-release or Preview build stands for Beta, and Stable its actually a release!  🍡 
that was actualy a reason for me to rename a branches, so then we migrate and sort issues related to that - Alfa willbe nightly and link to newest version (1.77.3 as per today) preview will be linked to 1.77.2, and stable will be 1.77.1 until we not happy to release 11.77.2 then its all will be slightly shifted

**Comment by simonai1254** [comment]
Just stumbled across https://github.com/mRemoteNG/mRemoteNG/issues/527#issuecomment-297805054, this might already lead into the right direction with multiple queries depending on what version is running. At the same time problems as described in #2541 should be handled better.

**Comment by simonai1254** [comment]
I just verified that if I download and run mRemoteNG 1.78.2-3207 and check for updates I still get "Update Available, Current Version is 1.76.11, you are running 1.78.2-3207") which is obviously incorrect. Therefore the update checker needs to be worked on so that works correctly and also takes the release channels properly into account.

**Comment by Kvarkas** [comment]
valid point

**Comment by p91paul** [comment]
For the time being, should we consider as the last stable 1.76.20 (the one marked as "Latest" release here on github) or 1.77.1 (that even if marked as "Pre-release" here on Github does not have the NB label in the release title)?
